### PR TITLE
feat: IFO Vesting

### PR DESCRIPTION
**Pull request by ChefMomota**

### CODE REVIEW

File: src/views/Ifos/components/IfoVesting/VestingPeriod/Claim.tsx
Comment (chefilip) [inline src/views/Ifos/components/IfoVesting/VestingPeriod/Claim.tsx]: should we use `useCallback`?

Comment (ChefMomota) [reply src/views/Ifos/components/IfoVesting/VestingPeriod/Claim.tsx]: Should we use? Cause i see Pool & farm not using `useCallback `

Comment (chefilip) [reply src/views/Ifos/components/IfoVesting/VestingPeriod/Claim.tsx]: For this case, it would not impact much but it's a good practice. Your call.

Comment (ChefMomota) [reply src/views/Ifos/components/IfoVesting/VestingPeriod/Claim.tsx]: Updated.

**Comment by chefilip** [comment]
LGTM

**Review by chefilip — APPROVED** [review]

File: src/views/Ifos/hooks/vesting/useFetchVestingData.ts
Comment (0xjojoex) [inline src/views/Ifos/hooks/vesting/useFetchVestingData.ts]: don't need first `await`, and consider use swr for this

File: src/views/Ifos/hooks/v3/useGetPublicIfoData.ts
Comment (0xjojoex) [inline src/views/Ifos/hooks/v3/useGetPublicIfoData.ts]: i'm partially ok with using bignumberjs for new feature, but why mixing both bignumber.js and ethers?

File: src/views/Ifos/hooks/v3/useGetWalletIfoData.ts
Comment (chef-jojo) [inline src/views/Ifos/hooks/v3/useGetWalletIfoData.ts:84]: same contract?

Comment (0xjojoex) [inline src/views/Ifos/hooks/v3/useGetWalletIfoData.ts]: `const contract = useIfoV3Contract(address)`
same contract above?

Comment (ChefMomota) [reply src/views/Ifos/hooks/v3/useGetWalletIfoData.ts:84]: Ya, is same. But i can't get read data from `const contract = useIfoV3Contract(address)`

File: src/views/Ifos/hooks/vesting/useFetchVestingData.ts
Comment (ChefMomota) [inline src/views/Ifos/hooks/vesting/useFetchVestingData.ts]: // TODO: revert back before merge.
const allVestingIfo: Ifo[] = ifosConfig.filter((ifo) => ifo.version >= 3.2 && ifo.vestingTitle && !ifo.isActive)

**Pull request merged**
